### ISSUE TITLE
Tooltip issue with TitleBar buttons

### ISSUE THREAD

**Issue by RaoHammas** [microsoft/microsoft-ui-xaml#2229]
Hi, I've noticed that the `ToolTip`  stays under the TitleBar's action Butons. It somehow looks weird and strange. I don't know if it's the default behavior and meant to be like this ? or maybe needs improvement ?

![toolTip Error](https://user-images.githubusercontent.com/16499626/78423973-b6546900-7683-11ea-865c-84dbe18d5cfa.gif)

**Comment by codendone** [comment]
Tooltips are normally hosted in a separate popup window which can draw on top of the min/max/close buttons.  You can see that behavior for the toolbar buttons in that area in Edge.

The question, then, is if there is something in your scenario which is preventing tooltips from using a popup window.  Is this in a UWP, or an island scenario?  Is there anything else you can think of which is special about your scenario?

**Comment by RaoHammas** [comment]
> Tooltips are normally hosted in a separate popup window which can draw on top of the min/max/close buttons. You can see that behavior for the toolbar buttons in that area in Edge.
> 
> The question, then, is if there is something in your scenario which is preventing tooltips from using a popup window. Is this in a UWP, or an island scenario? Is there anything else you can think of which is special about your scenario?


If you are asking me, It's happening in normal UWP application. I'm not sure if same happens in `Xaml Island` . Also in Edge tooltips always show below (bottom) according to what i just tried.

**Comment by RaoHammas** [comment]
> Maybe the tool tips should just open below? That way an inconsistency isn't created.
> 
> Edit: The ellipsis ...'s tooltip opens below at one point in the gif.

It opens below if mouse cursor is close to upper edge of button or element but if it's in middle or near bottom edge then tooltip automatically open above the element and in that case it stays below the action buttons.

**Comment by shaheedmalik** [comment]
> > Maybe the tool tips should just open below? That way an inconsistency isn't created.
> > Edit: The ellipsis ...'s tooltip opens below at one point in the gif.
> 
> It opens below if mouse cursor is close to upper edge of button or element but if it's in middle or near bottom edge then tooltip automatically open above the element and in that case it stays below the action buttons.

Not in gif.

**Comment by RaoHammas** [comment]
> > > Maybe the tool tips should just open below? That way an inconsistency isn't created.
> > > Edit: The ellipsis ...'s tooltip opens below at one point in the gif.
> > 
> > 
> > It opens below if mouse cursor is close to upper edge of button or element but if it's in middle or near bottom edge then tooltip automatically open above the element and in that case it stays below the action buttons.
> 
> Not in gif.

You can see that in gif too.

**Comment by shaheedmalik** [comment]
> > > > Maybe the tool tips should just open below? That way an inconsistency isn't created.
> > > > Edit: The ellipsis ...'s tooltip opens below at one point in the gif.
> > > 
> > > 
> > > It opens below if mouse cursor is close to upper edge of button or element but if it's in middle or near bottom edge then tooltip automatically open above the element and in that case it stays below the action buttons.
> > 
> > 
> > Not in gif.
> 
> You can see that in gif too.

I am referring to all of them up there.

**Comment by codendone** [comment]
@RaoHammas Is there anything special about your scenario or the version of Windows are you seeing this issue on?  Your gif either suggests windowed popups aren't being used, or that Windows is somehow confused in how it is drawing the min/max/close buttons.  Windowed popups should be used on any version of Windows 10 desktop, except when an app is running in Windows Mixed Reality.

I couldn't quite tell from your earlier comment: were you able to repro the same issue in Edge when the tooltip opened above the button into the mix/max/close area?

**Comment by RaoHammas** [comment]
@codendone sir nothing special. just a normal windows environment.
Edition : Windows 10 Pro
Version : 1909
OS build : 18363.752 

About Microsoft Edge  what i noticed is they never show tooltip above (they always show at bottom) that's why this issue is not happening with Edge.

**Comment by codendone** [comment]
@RaoHammas Were you reproing this using regular XAML (the version which is part of Windows), or was this using the WinUI 3 Alpha? The issue does repro in the WinUI 3 Alpha and in the Preview 1 released a couple of weeks ago, since those currently don't use the popup window support I mentioned previously. I'm still not able to repro this with regular XAML.

**Comment by RaoHammas** [comment]
> @RaoHammas Were you reproing this using regular XAML (the version which is part of Windows), or was this using the WinUI 3 Alpha? The issue does repro in the WinUI 3 Alpha and in the Preview 1 released a couple of weeks ago, since those currently don't use the popup window support I mentioned previously. I'm still not able to repro this with regular XAML.

Sir it was happening with WinUi 3 alpha. I don't know if same happens with regular Xaml.

**Comment by shaheedmalik** [comment]
> > it was happening with WinUi 3 alpha
> 
> That explains it! This should be fixed once WinUI 3 re-enables popup window support ([internal link](https://microsoft.visualstudio.com/DefaultCollection/OS/_workitems/edit/23865114)). If that doesn't happen for some reason, though, then we should do what is recommended above to have popup positioning explicitly avoid the min/max/close button area when the app is ExtendViewIntoTitleBar=true.

It happens on WinUI2+ too.

**Comment by bpulliam** [comment]
This has been fixed in WinAppSDK 1.3. (Task 26872199)
